doc/in-progress/object_versions.md

```diff
@@ -0,0 +1,146 @@
+
+Only an object's owner can equivocate it, but this is not a desirable thing to do.  You can avoid equivocation by carefully managing the versions of address-owned input objects.
+
+### Immutable Objects
+
+Like address-owned objects, you reference immutable objects at an ID and version, but they do not need to be locked as their contents and versions do not change. Their version is relevant because they could have started life as an address-owned object before being frozen. The given version identifies the point at which they became immutable.
+
+### Shared Objects
+
+Specifying a shared transaction input is slightly more complex. You reference it by its ID, the version it was shared at, and a flag indicating whether it is accessed mutably. You don't specify the precise version the transaction accesses because it is decided by consensus during transaction scheduling. When scheduling multiple transactions that touch the same shared object, validators agree the order of those transactions, and pick each transaction's input versions for the shared object accordingly (one transaction's output version becomes the next transaction's input version, and so on).
+
+Shared transaction inputs that you reference immutably participate in scheduling, but don't modify the object or increment its version.
+
+### Wrapped Objects
+
+The `make_wrapped` function in the following example creates an `Inner` object, wrapped in an `Outer` object, which is sent back to the transaction sender.
+
+```
+module example::wrapped {
+    use sui::object::{Self, UID};
+    use sui::transfer;
+    use sui::tx_context::{Self, TxContext};
+    
+    struct Inner has key {
+        id: UID,
+        x: u64,
+    }
+    
+    struct Outer has key {
+        id: UID,
+        inner: Inner,
+    }
+    
+    entry fun make_wrapped(ctx: &mut TxContext) {
+        transfer::transfer(
+            Outer {
+                id: object::new(ctx),
+                inner: Inner {
+                    id: object::new(ctx),
+                    x: 42,
+                }
+            }
+            tx_context::sender(ctx),
+        )
+    }
+}
+```
+
+You can't access wrapped objects by their ID in the object store, you must access them by the object that wraps them. In the prior example, the owner of `Outer` must specify it as the transaction input and then access its `inner` field to read the instance of `Inner`. Validators refuse to sign transactions that specify wrapped objects (like the `inner` of an `Outer`) as inputs.  As a result, you don't need to specify a wrapped object's version in a transaction that reads that object.
+
+Wrapped objects can eventually become "unwrapped", meaning that they are once again accessible at their ID:
```

Comment:
```suggestion
Wrapped objects are like deleted objects in that can't access wrapped objects by their ID in the object store, you must access them by the object that wraps them. In the prior example, the owner of `Outer` must specify it as the transaction input and then access its `inner` field to read the instance of `Inner`. Validators refuse to sign transactions that specify wrapped objects (like the `inner` of an `Outer`) as inputs.  As a result, you don't need to specify a wrapped object's version in a transaction that reads that object.

Unlike deleted objects, wrapped objects can eventually return to the object store, becoming "unwrapped". After this, they are once again accessible at their ID:
```

Reply:
Thoughts? I'm trying to put in some parallels to deleted objects but I'm not sure if it is actually helpful